lucet-wiggle/tests/wasi.rs

```diff
@@ -0,0 +1,456 @@
+        _oflags: types::Oflags,
+        _fs_rights_base: types::Rights,
+        _fs_rights_inherting: types::Rights,
+        _fdflags: types::Fdflags,
+    ) -> Result<types::Fd> {
+        unimplemented!("path_open")
+    }
+
+    fn path_readlink(
+        &self,
+        _fd: types::Fd,
+        _path: &GuestPtr<'_, str>,
+        _buf: GuestPtr<u8>,
+        _buf_len: types::Size,
+    ) -> Result<types::Size> {
+        unimplemented!("path_readlink")
+    }
+
+    fn path_remove_directory(&self, _fd: types::Fd, _path: &GuestPtr<'_, str>) -> Result<()> {
+        unimplemented!("path_remove_directory")
+    }
+
+    fn path_rename(
+        &self,
+        _fd: types::Fd,
+        _old_path: &GuestPtr<'_, str>,
+        _new_fd: types::Fd,
+        _new_path: &GuestPtr<'_, str>,
+    ) -> Result<()> {
+        unimplemented!("path_rename")
+    }
+
+    fn path_symlink(
+        &self,
+        _old_path: &GuestPtr<'_, str>,
+        _fd: types::Fd,
+        _new_path: &GuestPtr<'_, str>,
+    ) -> Result<()> {
+        unimplemented!("path_symlink")
+    }
+
+    fn path_unlink_file(&self, _fd: types::Fd, _path: &GuestPtr<'_, str>) -> Result<()> {
+        unimplemented!("path_unlink_file")
+    }
+
+    fn poll_oneoff(
+        &self,
+        _in_: GuestPtr<types::Subscription>,
+        _out: GuestPtr<types::Event>,
+        _nsubscriptions: types::Size,
+    ) -> Result<types::Size> {
+        unimplemented!("poll_oneoff")
+    }
+
+    fn proc_exit(&self, _rval: types::Exitcode) -> std::result::Result<(), ()> {
+        unimplemented!("proc_exit")
+    }
+
+    fn proc_raise(&self, _sig: types::Signal) -> Result<()> {
+        unimplemented!("proc_raise")
+    }
+
+    fn sched_yield(&self) -> Result<()> {
+        unimplemented!("sched_yield")
+    }
+
+    fn random_get(&self, _buf: GuestPtr<u8>, _buf_len: types::Size) -> Result<()> {
+        unimplemented!("random_get")
+    }
+
+    fn sock_recv(
+        &self,
+        _fd: types::Fd,
+        _ri_data: &types::IovecArray<'_>,
+        _ri_flags: types::Riflags,
+    ) -> Result<(types::Size, types::Roflags)> {
+        unimplemented!("sock_recv")
+    }
+
+    fn sock_send(
+        &self,
+        _fd: types::Fd,
+        _si_data: &types::CiovecArray<'_>,
+        _si_flags: types::Siflags,
+    ) -> Result<types::Size> {
+        unimplemented!("sock_send")
+    }
+
+    fn sock_shutdown(&self, _fd: types::Fd, _how: types::Sdflags) -> Result<()> {
+        unimplemented!("sock_shutdown")
+    }
+}
+
+/// Test the above generated code by running Wasm code that calls into it.
+#[test]
+fn main() {
+    // The `init` function ensures that all of the host call functions are linked into the
+    // executable.
+    crate::init();
+
+    // Temporary directory for outputs.
+    let workdir = TempDir::new().expect("create working directory");
+
+    // Build a C file into a Wasm module. Use the wasi-sdk compiler, but do not use the wasi libc
+    // or the ordinary start files,
```

Comment:
linewrap nit. sorry!

Reply:
sorry the vim wrapping command ive been using is under the extremely wrong impression that 100 cols is OK